File: Low Level Design (LLD)/Problems/06. Splitwise/Java (with Springboot)/splitwise/src/main/java/com/splitwise/splitwise/model/EqualExpense.java
package com.splitwise.splitwise.model;

import com.splitwise.splitwise.model.enums.ExpenseType;

import java.util.*;

public class EqualExpense extends Expense {
    public EqualExpense(String expenseId, String description, double amount, User paidBy, List<User> participants) {
        super(expenseId, description, amount, paidBy, participants, ExpenseType.EQUAL);
    }

    @Override
    public void calculateSplit() {
        double splitAmount = amount / participants.size();
        for (User user : participants) {
            userShare.put(user, splitAmount);
        }
    }
}

File: Low Level Design (LLD)/Problems/06. Splitwise/Cpp/model/src/EqualExpense.cpp
#include "../include/EqualExpense.h"

EqualExpense::EqualExpense(const std::string& expenseId,
                           const std::string& description,
                           double amount,
                           std::shared_ptr<User> paidBy,
                           const std::vector<std::shared_ptr<User>>& participants)
    : Expense(expenseId, description, amount, paidBy, participants, ExpenseType::EQUAL) {}

void EqualExpense::calculateSplit() {
    double splitAmount = amount / participants.size();
    auto& userShareMap = getUserShareMutable();

    for (const auto& user : participants) {
        userShareMap[user] = splitAmount;
    }
}
